### PR TITLE
make new module project contract

### PR DESCRIPTION
**Pull request by tarunnjoshi**
Targets #2252

### Description
Make new module Project Contract

### Checklist:

- [x] I have performed a self-review of my own code.

### CODE REVIEW

File: Modules/ProjectContract/Services/ProjectContractService.php [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+        $ProjectContractMeta->client_id = $request->get('client_id');
+        $ProjectContractMeta->website_url = $validated['website_url'];
+        $ProjectContractMeta->logo_img = $imageName;
+        $ProjectContractMeta->authority_name = $validated['authority_name'];
+        $ProjectContractMeta->contract_date_for_signing = $validated['contract_date_for_signing'];
+        $ProjectContractMeta->contract_date_for_effective = $validated['contract_date_for_effective'];
+        $ProjectContractMeta->contract_expiry_date = $validated['contract_expiry_date'];
+        $ProjectContractMeta->attributes = json_encode($request['attributes']);
+        $ProjectContractMeta->save();
```
Comment (gautam071) [inline Modules/ProjectContract/Services/ProjectContractService.php:32]: Use services!

File: Modules/ProjectContract/Services/ProjectContractService.php [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        $ProjectContractMeta->website_url = $validated['website_url'];
+        $ProjectContractMeta->logo_img = $validated['logo_img'];
+        $ProjectContractMeta->authority_name = $validated['authority_name'];
+        $ProjectContractMeta->contract_date_for_signing = $validated['contract_date_for_signing'];
+        $ProjectContractMeta->contract_date_for_effective = $validated['contract_date_for_effective'];
+        $ProjectContractMeta->contract_expiry_date = $validated['contract_expiry_date'];
+        $ProjectContractMeta->attributes = json_encode($request['attributes']);
+        $ProjectContractMeta->save();
+
+        return $ProjectContractMeta;
```
Comment (gautam071) [inline Modules/ProjectContract/Services/ProjectContractService.php:63]: here as well

**Review by gautam071 — CHANGES_REQUESTED** [review]
left some feedbacks

File: Modules/ProjectContract/Services/ProjectContractService.php (hunk 2 above)
Comment (tarunnjoshi) [reply Modules/ProjectContract/Services/ProjectContractService.php:63]: It is already correct .

File: Modules/ProjectContract/Resources/views/add-new-client.blade.php [hunk 3]
```diff
@@ -0,0 +1,103 @@
+@extends('layouts.app')
+
+@section('content')
+<div class=" ml-15">
+    <span class=""><h1>Add New Contract</h1></span>
```
Comment (gautam071) [inline Modules/ProjectContract/Resources/views/add-new-client.blade.php:5]: why span and h1 both?

File: Modules/ProjectContract/Resources/views/add-new-client.blade.php [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+</div><br>
```
Comment (gautam071) [inline Modules/ProjectContract/Resources/views/add-new-client.blade.php:6]:
```suggestion
</div>
<br>
```

File: Modules/ProjectContract/Resources/views/add-new-client.blade.php [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+                            placeholder="Enter Website url">
+                            <span class="text-danger">
+                                @error('website_url')
+                                {{$message}}
+                                @enderror
+                            </span>
+                        </div>
+                        <div class="form-group">
+                            <label class="field-required">Attributes:</label>
+                            <span data-toggle="tooltip" data-placement="right" title="Address, Summary, Authority Phone Number, Authority Email, Authority Designation." class="ml-2">
```
Comment (gautam071) [inline Modules/ProjectContract/Resources/views/add-new-client.blade.php:46]:
```suggestion
                            <span data-toggle="tooltip" data-placement="right" title="You can details like Address, Summary, Authority Phone Number, Authority Email, Authority Designation." class="ml-2">
```

File: Modules/ProjectContract/Resources/views/edit-project-contract.blade.php [hunk 6]
```diff
@@ -0,0 +1,103 @@
+@extends('layouts.app')
+
+@section('content')
+<div class=" ml-15">
+    <span class=""><h1>Edit Contract</h1></span>
+</div><br>
```
Comment (gautam071) [inline Modules/ProjectContract/Resources/views/edit-project-contract.blade.php:6]: same

File: Modules/ProjectContract/Resources/views/index.blade.php [hunk 7]
```diff
@@ -1,9 +1,45 @@
...
 @section('content')
-    <h1>Hello World</h1>
-
-    <p>
-        This view is loaded from module: {!! config('projectcontract.name') !!}
-    </p>
+<div class="d-flex justify-content-between align-items-center">
+    <h1>Project Contract</h1>
+    <a class="btn btn-success" href="{{route('projectcontract.view-form')}}"><i class="fa fa-plus mr-1" ></i>Add New Contract</a>
+</div><br>
```
Comment (gautam071) [inline Modules/ProjectContract/Resources/views/index.blade.php:7]:
```suggestion
</div>
<br>
```

File: Modules/ProjectContract/Resources/views/index.blade.php [hunk 8]
```diff
@@ -1,9 +1,45 @@
...
+            <td>{{$project->client->name}}</td>
+            <td><a href="{{$project->website_url}}"> <button class="btn btn-success fa fa mr-1  ">URL</button></td>
+                <td>{{$project->authority_name}}</td>
+                <td>{{$project->contract_date_for_signing}}</td>
+                <td>{{$project->contract_date_for_effective}}</td>
+                <td>{{$project->contract_expiry_date}}</td>
+                <td><img src="{{ asset('storage/contractlogo/'.$project->logo_img) }}" class="w-35 h-30 rounded-circle mb-1"></td>
+                <td>
+                    <a href="{{route('projectcontract.edit', $project->id)}}" class="pl-1 btn btn-link" ><i class="text-success fa fa-edit fa-lg"></i></a>
+                    <a href="{{route('projectcontract.delete', $project->id)}}" class="pl-1 btn btn-link" ><i class="text-danger fa fa-trash fa-lg"></i></a>
```
Comment (gautam071) [inline Modules/ProjectContract/Resources/views/index.blade.php:39]: Format

**Review by gautam071 — CHANGES_REQUESTED** [review]
Left some comments

File: Modules/ProjectContract/Resources/views/index.blade.php (hunk 8 above)
Comment (tarunnjoshi) [reply Modules/ProjectContract/Resources/views/index.blade.php]: Done with the changes.

File: Modules/ProjectContract/Resources/views/add-new-client.blade.php [hunk 9]
```diff
@@ -0,0 +1,104 @@
...
+                            placeholder="Enter Website url">
+                            <span class="text-danger">
+                                @error('website_url')
+                                {{$message}}
+                                @enderror
+                            </span>
+                        </div>
+                        <div class="form-group">
+                            <label class="field-required">Attributes:</label>
+                            <span data-toggle="tooltip" data-placement="right" title="You can details like Address, Summary, Authority Phone Number, Authority Email, Authority Designation." class="ml-2">
```
Comment (gautam071) [inline Modules/ProjectContract/Resources/views/add-new-client.blade.php:47]:
```suggestion
                            <span data-toggle="tooltip" data-placement="right" title="You can add details like Address, Summary, Authority Phone Number, Authority Email, Authority Designation." class="ml-2">
```

Comment (gautam071) [reply Modules/ProjectContract/Resources/views/add-new-client.blade.php:47]: same in edit as well

File: Modules/ProjectContract/Services/ProjectContractService.php [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+namespace Modules\ProjectContract\Services;
+
+use Modules\ProjectContract\Entities\ProjectContractMeta;
+use Modules\ProjectContract\Http\Requests\ProjectContractRequest;
+
+class ProjectContractService
+{
+    public function index()
+    {
+        return ProjectContractMeta::with('client')->orderBy('updated_at', 'desc')->get();
```
Comment (gautam071) [inline Modules/ProjectContract/Services/ProjectContractService.php:12]:
```suggestion
        return ProjectContractMeta::with('client')->orderBy('created_at', 'desc')->get();
```

**Review by gautam071 — COMMENTED** [review]
Looks good minor changes

File: Modules/ProjectContract/Services/ProjectContractService.php (hunk 10 above)
Comment (tarunnjoshi) [reply Modules/ProjectContract/Services/ProjectContractService.php]: Done with the changes

**Review by gautam071 — APPROVED** [review]
Looks good!

**Pull request merged**